### PR TITLE
TVB-2669: Review Forms

### PR DESCRIPTION
**Pull request by VinczeRobert**
As I said on Jira: "I replaced the usages of SimpleStrField with StrField and the usages of SimpleFloatField with FloatField and I made SimpleHiddenField inherit from TraitField. This means that only FormField remains which doesn't inherit from TraitField and we discussed with Paula that it should stay that way because it is different than the rest of them."

Furthermore, I think the issue described here https://github.com/the-virtual-brain/tvb-root/pull/235 is irrelevant because SimpleFloatField is gone and I don't see that problem in FloatField.

I still have to check carefully for duplicated code and as for not adding the form as a param on the fields I think it can be done, but then we need to send the project_id and the prefix as parameters and temporary files for TraitUploadField, draw_ranges for TraitDataTypeSelectField.

### CODE REVIEW

File: framework_tvb/tvb/core/neotraits/forms.py [hunk 1]
```diff
@@ -588,12 +503,10 @@ class Form(object):
     range_1 = None
```
Comment (popaula937) [inline framework_tvb/tvb/core/neotraits/forms.py:503]: Are there used?

File: framework_tvb/tvb/adapters/simulator/integrator_forms.py [hunk 2]
```diff
@@ -93,71 +93,71 @@ def fill_from_trait(self, trait):
 
 class HeunDeterministicIntegratorForm(IntegratorForm):
 
-    def __init__(self, prefix=''):
-        super(HeunDeterministicIntegratorForm, self).__init__(prefix)
+    def __init__(self):
```
[the remark this answers is not in the document]
Comment (VinczeRobert) [reply framework_tvb/tvb/adapters/simulator/integrator_forms.py:96]: I started deleting them, it's just that it requires a lot of other changes related to get_integrator_to_form_dict. I guess we can just replace that part with something like:

if isinstance(integrator, IntegratorStocastic):
     return IntegratorStochasticForm
else:
     return IntegratorForm

File: framework_tvb/tvb/adapters/simulator/monitor_forms.py [hunk 3]
```diff
@@ -180,91 +180,94 @@ def fill_from_trait(self, trait):
 
 class GlobalAverageMonitorForm(MonitorForm):
 
-    def __init__(self, session_stored_simulator=None, prefix='', project_id=None):
-        super(GlobalAverageMonitorForm, self).__init__(session_stored_simulator, prefix, project_id)
+    def __init__(self, session_stored_simulator=None, project_id=None):
```
[the remark this answers is not in the document]
Comment (VinczeRobert) [reply framework_tvb/tvb/adapters/simulator/monitor_forms.py:183]: Probably similar situation to what I wrote about integrators.

File: framework_tvb/tvb/adapters/simulator/integrator_forms.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (VinczeRobert) [reply framework_tvb/tvb/adapters/simulator/integrator_forms.py:96]: Never mind, done.

**Review by popaula937 — APPROVED** [review]

**Pull request merged**